src/rp2_common/pico_double/include/pico/double.h

```diff
@@ -16,64 +16,195 @@ extern "C" {
 *
-* On RP2350 the following additional functions are available; the _fast methods are faster but do not round correctly"
+* The user can control which version they want (e.g. **pico_double_xxx** by either setting the CMake global variable
+* `PICO_DEFAULT_DOUBLE_IMPL=xxx`, or by using the CMake function `pico_set_double_implementation(<TARGET> xxx)`. Note that in the absence
+* of either, pico_double_pico is used by default.
 *
-* - ddiv_fast, sqrt_fast
+* \if rp2040_specific
+* On RP2040, `pico_double_pico` uses optimized hand coded implementations from the bootrom and the SDK for both
+* basic double-precision floating point operations and floating point math library functions. These implementations
+* are generally faster and smaller than those provided by the C compiler/library, though they don't support all the features of a fully compliant
+* floating point implementation; they are however usually fine for the majority of cases
+* \endif
+*
+* \if rp2350_specific
+* On RP2350, `pico_double_pico` uses RP2350 DCP instructions (double co-processor) to implement fast version of the basic
+* arithmetic functions, and provides optimized M33 implementations of trignometric and scientific functions.
+* These implementations are generally faster and smaller than those provided by the C compiler/library, though they don't support all the features of a fully compliant
+* floating point implementation; they are however usually fine for the majority of cases
+* \endif
+*
+* On Arm, (replacement) optimized implementations are provided for the following compiler built-ins
+* and math library functions when using `pico_double_pico`:
+*
+* - basic arithmetic:
+*
+*   __aeabi_dadd, __aeabi_ddiv, __aeabi_dmul, __aeabi_drsub, __aeabi_dsub
+*
+* - comparison:
+*
+*   __aeabi_cfcmpeq, __aeabi_cfrcmple, __aeabi_cfcmple, __aeabi_dcmpeq, __aeabi_dcmplt, __aeabi_dcmple, __aeabi_dcmpge, __aeabi_dcmpgt, __aeabi_dcmpun
+*
+* - (u)int32 <-> double:
+*
+*    __aeabi_i2d, __aeabi_ui2d, __aeabi_d2iz, __aeabi_d2uiz
+*
+* - (u)int64 <-> double:
+*
+*   __aeabi_l2d, __aeabi_ul2d, __aeabi_d2lz, __aeabi_d2ulz
+*
+* - double -> float:
+*
+*   __aeabi_d2d
+*
+* - basic trigonometric:
+*
+*   sqrt, cos, sin, tan, atan2, exp, log
+*
+* - trigonometric and scientific
+*
+*   ldexp, copysign, trunc, floor, ceil, round, asin, acos, atan, sinh, cosh, tanh, asinh, acosh, atanh, exp2, log2, exp10, log10, pow, hypot, cbrt, fmod, drem, remainder, remquo, expm1, log1p, fma
+*
+* - GNU exetnsions:
+*
+*   powint, sincos
+*
+* On Arm, the following additional optimized functions are also provided when using `pico_double_pico`:
+*
+* - Conversions to/from integer types:
+*
+*   - (u)int -> double (round to nearest):
+*
+*     int2double, uint2double, int642double, uint642double
+*
+*     note: on `pico_double_vfp` the 32-bit functions are also provided as C macros too since they map to inline VFP code
```

Comment:
There's still several mentions of `pico_double_vfp`, which (unlike `pico_float_pico_vfp`) isn't listed earlier as one of the available pico_double "flavours"?